[BUG] Makefile works fail on go 1.18

**What happened**:
In my env, the go version is `go 1.18.2`, when do the command `make manifests` in the master branch, got the error:
```
make manifests

rm -rf /Users/xxx/kruise/bin/controller-gen
go: creating new go.mod: module tmp
Downloading sigs.k8s.io/controller-tools/cmd/controller-gen@v0.7.0
go: added github.com/fatih/color v1.12.0
...
go: added sigs.k8s.io/controller-tools v0.7.0
go: added sigs.k8s.io/structured-merge-diff/v4 v4.1.2
go: added sigs.k8s.io/yaml v1.2.0
/Users/xxx/kruise/bin/controller-gen "crd:crdVersions=v1" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
bash: /Users/xxx/kruise/bin/controller-gen: No such file or directory
make: *** [manifests] Error 127
```
After check, it is the `controller-gen` binary not generated by `go get` in go 1.18.

**What you expected to happen**:
To make the Makefile both work well on go version below or above go 1.18

**How to reproduce it (as minimally and precisely as possible)**:
1. Git clone the kruise master branch.
2. Set the go version with go 1.18.
3. Do `make manifests` or `make generate manifests`, then will get the error.

**Anything else we need to know?**:

**Environment**:
- OS: Mac
- Go: go 1.18.2

--- a/Makefile
+++ b/Makefile
@@ -91,7 +91,7 @@ endif
 
 KUSTOMIZE = $(shell pwd)/bin/kustomize
 kustomize: ## Download kustomize locally if necessary.
-	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v3.8.7)
+	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v4@v4.5.5)
 
 GOLANGCI_LINT = $(shell pwd)/bin/golangci-lint
 golangci-lint: ## Download golangci-lint locally if necessary.
@@ -110,7 +110,7 @@ TMP_DIR=$$(mktemp -d) ;\
 cd $$TMP_DIR ;\
 go mod init tmp ;\
 echo "Downloading $(2)" ;\
-GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
+GOBIN=$(PROJECT_DIR)/bin go install $(2) ;\
 rm -rf $$TMP_DIR ;\
 }
 endef

